### ISSUE TITLE
Continuation of Postal

### ISSUE THREAD

**Issue by waaghals** [andrewdavey/postal#172]
Postal (at least in this repository) is no longer actively maintained (last release sept 2014) and is not accepting any new PR's. My initial thought was forking it and maintaining Postal myself.
Luckily some users already started doing this.
I see at least two active forks at the moment; [hermanho's](https://github.com/hermanho/postal) and [John Harman's](https://github.com/Jmaharman/postal/network)

Both these forks have much needed features like [.Net Core](https://github.com/andrewdavey/postal/issues/156) support and [layout support outside of MVC](https://github.com/andrewdavey/postal/pull/117). However these features are not available in all forks and make for some fragmentation, these forks also have no published packages (as far as I'm aware)  making them hard to use. 

I'm using John Harman's fork because I've needed layouts outside of MVC, but using this was a pain because I needed to create a private package myself.

@andrewdavey , @hermanho and @JmaHarman
I wanted to open the discussion about the continuation of Postal.
Merge different forks into a singular maintained Postal and start accepting PRs again.

What are your thoughts/ideas for continuation of Postal?

**Comment by hermanho** [comment]
It sound good. I am working my fork on asp.net core support and you can find the package in nuget https://www.nuget.org/packages/Postal.AspNetCore/ ,

**Comment by andrewdavey** [comment]
I no longer have the time to work on Postal. Would someone else like to take ownership? It would be great to see the project continue.

**Comment by Jmaharman** [comment]
I don't have time in the evenings / weekends to do anything on it, but as
it comes up at work we're more than  happy to put time towards it.

We took the source across to bitbucket into a private repo because the
source started diverging a lot from what would be useful publicly.

However there are some commits in there we would likely find useful, i.e.
updating RazorEngine to facilitate _Layouts being used when outside of MVC.

I'm not up to speed on whether we can manage both a .NET Core package with
the new features mentioned above, while still maintaining it for .NET 4.5
but I'm sure one of you clever gentlemen would know more about that.

To summarise, I'm happy to either provide our changes to you so you can
work out how to apply them in the public github repo, or do a few PRs to
bring those features in myself.

John

**Comment by waaghals** [comment]
Thanks for all the quick replies.

> I don't have time in the evenings / weekends to do anything on it, but as
it comes up at work we're more than happy to put time towards it.

It is the same for me, I won't have much time in terms of continuous development. However if it is any value for my employer than I could find some time to review/merge any PR or contribute some fixes.

I don't think continuous development is needed. But reviewing PR and merging any contributes could already be a great improvement towards continuing Postal.

I'm willing to spend some time to help, but I'm unsure how much time I'm able to invest. I was hoping maybe we could come up with a solution with multiple maintainer sharing any load.

**Comment by daedalus-codes** [comment]
I'm interested, though I'm swamped with work for the next few months.  However, we use this pretty heavily in our MVC projects.  

I was considering porting it to use MailKit/MimeKit but still be able to use System.Web.Mail in another Nuget package.  Just some thoughts I've had when using it over the years.

**Comment by hermanho** [comment]
Is it possible to setup a CI (AppVeyor, TravisCI) with some test cases to improve the PR reviewing and merging ?

**Comment by phantomtypist** [comment]
I have some bandwidth.

**Comment by andrewdavey** [comment]
I've added @phantomtypist as a collaborator. I'm happy to hand over the reins! :)

**Comment by natiki** [comment]
@phantomtypist Is Postal continuing?

**Comment by phantomtypist** [comment]
Yes, yes indeed!  I'm not planning on doing anything major until after the New Year though.  I'll be doing some back-end prep work between now and then working with @andrewdavey, but nothing serious.

**Comment by natiki** [comment]
Cool!!!!! Hope to see the other repos consolidated back to here.

**Comment by phantomtypist** [comment]
@natiki Forgive my ignorance, but what other repos are you talking about?  Are you talking about the forked repos and "forked" unofficial nuget packages?

**Comment by natiki** [comment]
@phantomtypist Yes I was referring to the forks mentioned in the first entry of this thread https://github.com/andrewdavey/postal/issues/172#issue-224031417. I don't need the .Core support but upgrading to the latest Razor Engine and support for layouts would be great.

**Comment by natiki** [comment]
@phantomtypist Was hoping we might have a new year's surprise. Ultimately I am looking for layout support in Postal which appears to be ready to go in https://github.com/andrewdavey/postal/pull/117 Any ideas of when the next release will be out and what will be in it?

**Comment by phantomtypist** [comment]
@natiki I'm not dead!

In all seriousness I'm ready to begin working on this project next week.  I work at a uni and it's been a busy Winter session.  Winter and Summer are the busiest times for me because nobody else is around so to speak.

Plan of attack is scaffold the CI/CD pipeline and then I'll triage the issues... and since you mentioned it, I'll look at #117 first.

**Comment by natiki** [comment]
@phantomtypist Thanks for that. Not stalking you ;-) Hopefully an upgrade to the Razor engine will help with a few of the issues in the list.

**Comment by rsambhangi** [comment]
@phantomtypist We decided to make use of Postal for sending transaction emails using hangfire.io in our windows service app. Can't wait to see layout support in Postal for non asp.net application with upgraded Razor engine. Please let me know when are you planning to release. Thank you!

**Comment by phantomtypist** [comment]
I wanted to provide an update.  I _promise_ I'll start working on this over the weekend.  I've been sidetracked by stuff at work way longer than anticipated :(, but I'm wrapping up the important things tomorrow.

**Comment by p10tyr** [comment]
If somebody is interested in just generating Razor (With strong typed models) to HTML then look at this gist
It applies to NetCore 2+ and works on Linux backed dockers
https://gist.github.com/ppumkin/9ca1ef2322f1390a6d5cda438e5e2319

**Comment by dlidstrom** [comment]
Hi guys, I’m still using this fine library. Have any of you moved on to other libraries? What is recommended these days?

**Comment by rickdgray** [comment]
@andrewdavey, phantom has not come through and done anything with this project. I see a .NET 5 migration PR done [here](https://github.com/andrewdavey/postal/pull/181) by @johanjvr. I also see that @hermanho has continued [his branch](https://github.com/hermanho/postal.aspnetcore) and is still maintaining it. Might I recommend giving them collaborator permissions instead?

**Comment by rickdgray** [comment]
I would be happy to go through the issues and PRs myself and play catch up too if you would like to give me permissions.

**Comment by andrewdavey** [comment]
I've added you as a collaborator @rickdgray - happy for you to help out!

**Comment by darronclarison412** [comment]
It's essential to consider consolidating the active forks of Postal to ensure a unified, maintained version. Merging these forks and encouraging PRs could greatly benefit the [Recliners](https://allaboutrecliners.com/) community. Communication and collaboration among contributors are key for the continued success and evolution of Postal.

**Comment by rickdgray** [comment]
Apologies for not yet following up. I am in grad school and the past few semesters have taken significantly more time than expected. I anticipate moving forward on PRs in a few months after I graduate.

**Comment by gladstons** [comment]
Great initiative in considering the future of Postal! Consolidating active forks into a singular, maintained version seems like a pragmatic approach. It would streamline features and [Slide ](https://slidesharedown.com/)enhance usability.
